internal/mapper/datasource_mapper.go

```diff
@@ -141,18 +141,21 @@ func generateDataSourceSchema(logger *slog.Logger, name string, dataSource explo
 				continue
 			}
 
-			parameterAttribute, schemaErr := s.BuildDataSourceAttribute(paramName, computability)
+			paramPath := strings.Split(paramName, ".")
+
+			parameterAttribute, schemaErr := s.BuildDataSourceAttribute(paramPath[len(paramPath)-1], computability)
 			if schemaErr != nil {
 				log.WarnLogOnError(pLogger, schemaErr, "skipping mapping of read operation parameter")
 				continue
 			}
 
-			readParameterAttributes = append(readParameterAttributes, parameterAttribute)
+			// TODO: currently, no errors can be returned from merging, but in the future we should consider raising errors/warnings for unexpected scenarios, like type mismatches between attribute schemas
+			readResponseAttributes, _ = readResponseAttributes.MergeAttribute(paramPath, parameterAttribute, computability)
 		}
 	}
 
 	// TODO: currently, no errors can be returned from merging, but in the future we should consider raising errors/warnings for unexpected scenarios, like type mismatches between attribute schemas
-	dataSourceAttributes, _ := readParameterAttributes.Merge(readResponseAttributes)
+	dataSourceAttributes := readResponseAttributes
```

Comment:
I think we can remove this renaming of the variable and maybe just change `readResponseAttributes` -> `dataSourceAttributes` declared on line 85